### ISSUE TITLE
Logback with custom configuration crashes with Log levels defined in application.yml 

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#8678]
- [ ] Create a Micronaut Application

- [ ] remove `src/main/resources/logback.xml`

- [ ] Change `logback` dependency from:

`runtimeOnly("ch.qos.logback:logback-classic")`

to : 

`implementation("ch.qos.logback:logback-classic")`

- [ ] Create file `src/main/resources/META-INF/services/ch.qos.logback.classic.spi.Configurator`

with content: 

```
example.micronaut.CustomConfigurator
```

- [ ] Add a [Custom configuration](https://logback.qos.ch/manual/configuration.html#auto_configuration):

```java
package example.micronaut;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.Logger;
import ch.qos.logback.classic.LoggerContext;
import ch.qos.logback.classic.layout.TTLLLayout;
import ch.qos.logback.classic.spi.Configurator;
import ch.qos.logback.classic.spi.ILoggingEvent;
import ch.qos.logback.core.ConsoleAppender;
import ch.qos.logback.core.encoder.LayoutWrappingEncoder;
import ch.qos.logback.core.spi.ContextAwareBase;

/**
 * @see <a href="https://github.com/qos-ch/logback/blob/master/logback-classic/src/main/java/ch/qos/logback/classic/BasicConfigurator.java">Logback Class Basic Configurator</a>
 */
public class CustomConfigurator extends ContextAwareBase implements Configurator {

    public static final Level ROOT_LOGGER_LEVEL = Level.INFO;

    @Override
    public void configure(LoggerContext lc) {
        addInfo("Setting up default configuration.");
        ConsoleAppender<ILoggingEvent> ca = startConsoleAppender(lc);
        configureRootRootLogger(lc, ca);

        final String pkg = "example.micronaut";
        Logger appPkgLogger = lc.getLogger(pkg);
        appPkgLogger.setLevel(Level.TRACE);
        appPkgLogger.setAdditive(false);
        appPkgLogger.addAppender(ca);

        Logger controllersLogger = lc.getLogger(pkg + ".controllers");
        controllersLogger.setLevel(Level.INFO);
        controllersLogger.setAdditive(false);
        controllersLogger.addAppender(ca);
        
        Logger mnLogger = lc.getLogger("io.micronaut.runtime.Micronaut");
        mnLogger.setLevel(Level.INFO);
        mnLogger.setAdditive(false);
        mnLogger.addAppender(ca);
    }

    private void configureRootRootLogger(LoggerContext lc, ConsoleAppender<ILoggingEvent> ca) {
        Logger rootLogger = lc.getLogger(Logger.ROOT_LOGGER_NAME);
        rootLogger.setLevel(ROOT_LOGGER_LEVEL);
        rootLogger.addAppender(ca);
    }

    private ConsoleAppender<ILoggingEvent> startConsoleAppender(LoggerContext lc) {
        ConsoleAppender<ILoggingEvent> ca = new ConsoleAppender<>();
        ca.setContext(lc);
        ca.setName("stdout");
        LayoutWrappingEncoder<ILoggingEvent> encoder = new LayoutWrappingEncoder<>();
        encoder.setContext(lc);

        // same as
        // PatternLayout layout = new PatternLayout();
        // layout.setPattern("%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} -
        // %msg%n");
        TTLLLayout layout = new TTLLLayout();

        layout.setContext(lc);
        layout.start();
        encoder.setLayout(layout);

        ca.setEncoder(encoder);
        ca.start();
        return ca;
    }
}
```

- [ ] Replace `src/main/java/example/micronaut/Application.java`

```java
package example.micronaut;

import io.micronaut.runtime.Micronaut;
import org.slf4j.LoggerFactory;
import org.slf4j.Logger;

public class Application {
    private static final Logger LOG = LoggerFactory.getLogger(Application.class);

    public static void main(String[] args) {
        LOG.trace("starting the app");
        Micronaut.run(Application.class, args);
    }
}
```

- [ ] Add [Logging configuration](https://docs.micronaut.io/latest/guide/#loggingConfiguration).

```yaml
---
logger:
  levels:
    example.micronaut.controllers: TRACE
```

- [ ] Run the application

```
./gradlew run
```

- [ ] Do the following curl request

`curl localhost:8080`

- [ ] Verify yo use in the logs

```
TRACE example.micronaut.controllers.HomeController - inside hello world
```

## Actual Behaviour

It fails to start

## Expected Behaviour 

It works 

# version

Micronaut Framework 3.8.2

# sample application

[https://github.com/grails-core-issues-forks/micronaut-core-8678](https://github.com/grails-core-issues-forks/micronaut-core-8678)

**Comment by sdelamo** [comment]
@timyates close this issue if it is resolved.

**Comment by timyates** [comment]
Fixed in 3.8.4 by #8674
